### PR TITLE
Custom level

### PR DESCRIPTION
**Pull request by erikajoun**
Here are the list of things I did:
-Added a new castle level, which is in the CastleLevel scene.
-Added the boolean "invincible" to the shadow player to make testing levels easier.
-Added ComboInput and ComboUnlockable scripts
-Removed excess unityevents in editor from the Button prefab

### CODE REVIEW

File: Assets/Scripts/ComboInput.cs [hunk 1]
```diff
@@ -0,0 +1 @@
...
        for (int i = 0; i < objects.Length; i++)
        {
            ComboUnlockable script = objects[i].GetComponent<ComboUnlockable>();
            if (script != null)
            {
                script.Subtract(value);
            }
        }
    }
}
```
Comment (sru) [inline Assets/Scripts/ComboInput.cs]: Something's wrong with this file. Looks like there is no new lines. Can you fix it?

File: Assets/Scripts/ComboUnlockable.cs [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
using UnityEngine.Events;

/**
* Works with the ComboInput script to allow "unlockable" objects to
* perform an action as a result of the simultaneous activation of multiple 
* "input" objects. Attach this script to all "unlockable" objects that need to
* be triggered by multiple objects.
*/
public class ComboUnlockable : MonoBehaviour
{
```
Comment (sru) [inline Assets/Scripts/ComboUnlockable.cs]: Same as `ComboInput.cs`. Add new lines.

File: Assets/Scripts/PlayerController.cs [hunk 3]
```diff
@@ -43,6 +43,9 @@ public enum PlayerType { Light, Shadow };
     // The script from the interactable object
     private Interactable interactableScript = null;
 
+    // For debugging purposes
+    public bool invincible = false;
```
Comment (sru) [inline Assets/Scripts/PlayerController.cs]: Better way would be separating out the dying mechanism to another component. That way, we can just disable or enable those components containing dying mechanism to debug and test.

File: Assets/Scripts/Rideable.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Rideable : MonoBehaviour {
```
Comment (sru) [inline Assets/Scripts/Rideable.cs]: What does this do? What is it for?

**Review by sru — CHANGES_REQUESTED** [review]
I haven't really looked at the level yet, just codes.

File: Assets/Scripts/ComboInput.cs (hunk 1 above)
Comment (erikajoun) [reply Assets/Scripts/ComboInput.cs]: That is very weird... I have just fixed both of the scripts.

**Comment by sru** [comment]
I just played the level with my brother. We concluded that the puzzle itself was fine, but the platforms were so close together that we found it difficult to determine which ones we can jump over and which ones we cannot jump over. I felt that the characters should be able to jump over all the platforms, but no, I can't really jump onto any platforms without help of the boxes.

**Comment by sru** [comment]
Oh, and remove `Assets/Sprites/Objects/GrowingPlant.meta`. The branch does not have anything committed named `Assets/Sprites/Objects/GrowingPlant`, but the meta file is still committed. You should remove it (or, not include it in the commits).

**Pull request merged**